aiohttp/web_ws.py

```diff
@@ -495,7 +495,11 @@ async def close(
 
         try:
             async with async_timeout.timeout(self._timeout):
-                msg = await reader.read()
+                while True:
+                    msg = await reader.read()
+                    if msg.type is WSMsgType.CLOSE:
```

Comment:
We might get WSMsgType.ERROR as well

Reply:
I reraise `msg.data` from here, so it's handled on the exception path.

Reply:
I think that should be fine. Will need a test for this case as well

Reply:
Interestingly the `WSMsgType.ERROR` case seems to be already covered. The missing branch coverage is (probably) actually for the situation that this PR is meant to solve, which is when a regular message is received by the server within the close window. I added this test now.